src/utils/enterprise-contract-utils.tsx

```diff
@@ -45,7 +45,11 @@ export const getRuleStatus = (type: ENTERPRISE_CONTRACT_STATUS) => {
 };
 
 export const extractEcResultsFromTaskRunLogs = (logs: string): EnterpriseContractResult => {
-  const extractedLogs = logs.match(/(\[report-json\] ).+/g);
-  const json = JSON.parse(extractedLogs.map((l) => l.replace('[report-json] ', '')).join(''));
-  return json;
+  const EC_REPORT_JSON_REGEX = /((?<=step-report-json\s*:-\s*)(\{.*?\})(?=\s*step-|$))/g;
```

Comment:
Changes of utils is duplicated with #84. :-). We can focus on changes of the component.